apollo-cache-interceptor/src/commonMain/kotlin/com/apollographql/apollo/interceptor/cache/ApolloCacheInterceptor.kt

```diff
@@ -0,0 +1,95 @@
+package com.apollographql.apollo.interceptor.cache
+
+import com.apollographql.apollo.api.ApolloExperimental
+import com.apollographql.apollo.api.Operation
+import com.apollographql.apollo.api.Response
+import com.apollographql.apollo.api.Response.Companion.builder
+import com.apollographql.apollo.api.ResponseField
+import com.apollographql.apollo.api.internal.NoOpResolveDelegate
+import com.apollographql.apollo.cache.CacheHeaders
+import com.apollographql.apollo.cache.normalized.CacheKey
+import com.apollographql.apollo.cache.normalized.CacheKeyResolver
+import com.apollographql.apollo.cache.normalized.Record
+import com.apollographql.apollo.cache.normalized.internal.CacheFieldValueResolver
+import com.apollographql.apollo.cache.normalized.internal.CacheKeyBuilder
+import com.apollographql.apollo.cache.normalized.internal.ReadableStore
+import com.apollographql.apollo.cache.normalized.internal.RealCacheKeyBuilder
+import com.apollographql.apollo.cache.normalized.internal.ResponseNormalizer
+import com.apollographql.apollo.cache.normalized.simple.SimpleNormalizedCache
+import com.apollographql.apollo.interceptor.ApolloInterceptorChain
+import com.apollographql.apollo.interceptor.ApolloRequest
+import com.apollographql.apollo.interceptor.ApolloRequestInterceptor
+import com.apollographql.apollo.interceptor.ApolloResponse
+import com.apollographql.apollo.internal.response.RealResponseReader
+import com.apollographql.apollo.internal.response.RealResponseWriter
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+
+@ApolloExperimental
+class ApolloCacheInterceptor : ApolloRequestInterceptor {
+  val normalizedCache = SimpleNormalizedCache()
+  val readableStore = object : ReadableStore {
+    override fun read(key: String, cacheHeaders: CacheHeaders): Record? {
+      return normalizedCache.loadRecord(key, cacheHeaders)
+    }
+
+    override fun read(keys: Collection<String>, cacheHeaders: CacheHeaders): Collection<Record> {
+      return keys.mapNotNull { normalizedCache.loadRecord(it, cacheHeaders) }
+    }
+  }
+
+  override fun <D : Operation.Data> intercept(request: ApolloRequest<D>, chain: ApolloInterceptorChain): Flow<ApolloResponse<D>> {
+    return flow {
+      val response = readFromCache(request)
+      if (response != null) {
+        emit(ApolloResponse(requestUuid = request.requestUuid, response = response, executionContext = request.executionContext + CacheExecutionContext(true)))
```

Comment:
nit: could you pls put arguments on separate lines

Reply:
Sure thing!